daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Classify.scala

```diff
@@ -25,15 +25,19 @@ private[speedy] object Classify { // classify the machine state w.r.t what step
   }
 
   def classifyMachine(machine: Machine, counts: Counts): Unit = {
-    if (machine.returnValue != null) {
-      // classify a value by the continution it is about to return to
-      counts.ctrlValue += 1
-      val kont = machine.kontStack.get(machine.kontStack.size - 1).getClass.getSimpleName
-      val _ = counts.konts += kont -> (counts.konts.get(kont).getOrElse(0) + 1)
-    } else {
-      counts.ctrlExpr += 1
-      val expr = machine.ctrl.getClass.getSimpleName
-      val _ = counts.exprs += expr -> (counts.exprs.get(expr).getOrElse(0) + 1)
+    machine.control match {
+      case Control.WeAreUnset() => ()
+      case Control.WeAreComplete() => ()
+      case Control.WeAreHungry(_) => ()
+      case Control.Value(_) =>
+        // classify a value by the continution it is about to return to
+        counts.ctrlValue += 1
+        val kont = machine.kontStack.get(machine.kontStack.size - 1).getClass.getSimpleName
+        val _ = counts.konts += kont -> (counts.konts.get(kont).getOrElse(0) + 1)
+      case Control.Expression(exp) =>
+        counts.ctrlExpr += 1
+        val expr = exp.getClass.getSimpleName
+        val _ = counts.exprs += expr -> (counts.exprs.get(expr).getOrElse(0) + 1)
```

Comment:
Nit pick: would there be any value in simplifying with a catch all case like:
```
machine.control match {
  case Control.Value(_) => ...
  case Control.Expression(exp) => ...
  case _ => ()
}
```

Reply:
I'm not a big fan of catch all matches. maybe would be ok here.
(but as I wrote above, these error cases might not survive my future refactorings!)